hack/release/common.sh

```diff
@@ -66,136 +53,165 @@ authenticatePrivateRepo() {
-  fi
+  date_epoch="$(dateEpoch)"
+  build_date="$(buildDate "${date_epoch}")"
+
+  img="$(GOFLAGS=${GOFLAGS:-} SOURCE_DATE_EPOCH="${date_epoch}" KO_DATA_DATE_EPOCH="${date_epoch}" KO_DOCKER_REPO="${oci_repo}" ko publish -B -t "${version}" ./cmd/controller)"
+  img_repo="$(echo "${img}" | cut -d "@" -f 1 | cut -d ":" -f 1)"
+  img_tag="$(echo "${img}" | cut -d "@" -f 1 | cut -d ":" -f 2 -s)"
+  img_digest="$(echo "${img}" | cut -d "@" -f 2)"
+
+  cosignOciArtifact "${commit_sha}" "${build_date}" "${img}"
+
+  yq e -i ".controller.image.repository = \"${img_repo}\"" charts/karpenter/values.yaml
+  yq e -i ".controller.image.tag = \"${img_tag}\"" charts/karpenter/values.yaml
+  yq e -i ".controller.image.digest = \"${img_digest}\"" charts/karpenter/values.yaml
+
+  publishHelmChart "${oci_repo}" "karpenter" "${helm_chart_version}" "${commit_sha}" "${build_date}"
+  publishHelmChart "${oci_repo}" "karpenter-crd" "${helm_chart_version}" "${commit_sha}" "${build_date}"
 }
 
-helmChartVersion(){
-    RELEASE_VERSION=$1
-    if [[ $(releaseType "$RELEASE_VERSION") == "$RELEASE_TYPE_STABLE" ]]; then
-      echo "${RELEASE_VERSION#v}"
-    fi
+publishHelmChart() {
+  local oci_repo helm_chart version commit_sha build_date ah_config_file_name helm_chart_artifact helm_chart_digest
+
+  oci_repo="${1}"
+  helm_chart="${2}"
+  version="${3}"
+  commit_sha="${4}"
+  build_date="${5}"
+
+  ah_config_file_name"${helm_chart}/artifacthub-repo.yaml"
+  helm_chart_artifact="${helm_chart}-${version}.tgz"
 
-    if [[ $(releaseType "$RELEASE_VERSION") == "$RELEASE_TYPE_SNAPSHOT" ]]; then
-      echo "${CURRENT_MAJOR_VERSION}-${RELEASE_VERSION}"
-    fi
+  yq e -i ".appVersion = \"${version}\"" "charts/${helm_chart}/Chart.yaml"
+  yq e -i ".version = \"${version}\"" "charts/${helm_chart}/Chart.yaml"
+
+  cd charts
+  [[ -s "${ah_config_file_name}" ]] && oras push "${oci_repo}:artifacthub.io" --config /dev/null:application/vnd.cncf.artifacthub.config.v1+yaml "${ah_config_file_name}:application/vnd.cncf.artifacthub.repository-metadata.layer.v1.yaml"
+  helm dependency update "${helm_chart}"
+  helm lint "${helm_chart}"
+  helm package "${helm_chart}" --version "${version}"
+  helm push "${helm_chart_artifact}" "oci://${oci_repo}"
+  rm "${helm_chart_artifact}"
+  cd ..
+
+  helm_chart_digest="$(crane digest "${oci_repo}:${version}")"
+  cosignOciArtifact "${commit_sha}" "${build_date}" "${oci_repo}:${version}@${helm_chart_digest}"
 }
 
-buildDate(){
-    # Set the SOURCE_DATE_EPOCH and KO_DATA_DATE_EPOCH values for reproducable builds with timestamps
-    # https://ko.build/advanced/faq/
-    DATE_FMT="+%Y-%m-%dT%H:%M:%SZ"
-    SOURCE_DATE_EPOCH=$(git log -1 --format='%ct')
-    echo "$(date -u -r "${SOURCE_DATE_EPOCH}" $DATE_FMT 2>/dev/null)"
+cosignOciArtifact() {
+  local commit_sha build_date artifact
+
+  commit_sha="${1}"
+  build_date="${2}"
+  artifact="${3}"
+
+  cosign sign --yes -a commitSha="${commit_sha}" -a buildDate="${build_date}" "${artifact}"
```

Comment:
Seems like we aren't attesting the GIT_VERSION anymore?

Reply:
The git version was being added as an annotation; annotations aren't checked as part of validation unless opted in to. I can add a `version` attribute which can be checked as part of verification if there is the need to correlate the SHA with a version, although I'm not sure that's necessary.

Reply:
I've added the `version` attribute.